src/python/zquantum/core/circuit/_gate_test.py

```diff
@@ -1,7 +1,184 @@
+                ("CZ", None),
+                ("Ry", [-pi + 0.5]),
+                ("Rx", [-pi]),
+                ("CZ", None),
+                ("Rx", [pi / 2]),
+                ("Rz", [0.5]),
+            ]
+
+            # When
+            if gate_name == "CPHASE":
+                cphase_decomposition = gate.to_qiskit(qreg=qreg)
+                for i in range(int(len(cphase_decomposition) / 3)):
+                    current_gate = cphase_decomposition[3 * i]
+                    current_qreg = cphase_decomposition[3 * i + 1]
+                    target_gate, qubit_list = self.create_gate(
+                        cphase_targets[i][0], params=cphase_targets[i][1]
+                    )
+                    recreated_current_gate = Gate.from_qiskit(current_gate, qubit_list)
+                    self.assertEqual(target_gate, recreated_current_gate)
+                continue
+
+            else:
+                qiskit_gate, qreg, creg = gate.to_qiskit(qreg=qreg)
+                recreated_gate = Gate.from_qiskit(qiskit_gate, qubit_list)
+
+            # Then
+            if gate_name == "PHASE":
+                rz_gate, _ = self.create_gate("Rz")
+                self.assertEqual(rz_gate, recreated_gate)
+            else:
+                self.assertEqual(gate, recreated_gate)
+
+    def test_dict_io_for_symbolic_parameters(self):
+        for gate_name in self.one_parameter_gates:
+            # Given
+            gate, _, _ = self.create_gate_with_symbolic_params(gate_name)
+
+            # When
+            gate_dict = gate.to_dict()
+            recreated_gate = Gate.from_dict(gate_dict)
+
+            # Then
+            self.assertEqual(gate, recreated_gate)
+
+    def test_pyquil_io_for_symbolic_parameters(self):
+        for gate_name in self.one_parameter_gates:
+            # Given
+            gate, qubit_list, _ = self.create_gate_with_symbolic_params(gate_name)
+
+            # When
+            pyquil_gate = gate.to_pyquil()
+            recreated_gate = Gate.from_pyquil(pyquil_gate, qubit_list)
+
+            # Then
+            self.assertEqual(gate, recreated_gate)
+
+    def test_cirq_io_for_symbolic_parameters(self):
+        for gate_name in self.one_parameter_gates:
+            # Given
+            gate, qubit_list, params = self.create_gate_with_symbolic_params(gate_name)
+            symbols_map = []
+            for param in params:
+                symbols_map.append((param, 0.38553))
+
+            # When
+            cirq_gate = gate.to_cirq()
+            recreated_gate = Gate.from_cirq(cirq_gate, qubit_list)
+            gate_evaluated = gate.create_evaluated_gate(symbols_map)
+            recreated_gate_evaluated = recreated_gate.create_evaluated_gate(symbols_map)
+
+            # Then
+            # There were numerical & sympy related issues when comparing gates directly, so in this case we compare the evaluated forms of the gates.
+            self.assertEqual(gate_evaluated, recreated_gate_evaluated)
```

Comment:
Because of this, I think we should test a couple different assignments for the parameter. Including corner cases such as `0`, `pi`, `-pi`, `2pi`, etc

Reply:
Done in `26e22a34f6d398f0145ca301500ecb2964a0dc40`